core/test/scripts/i18n/collect-strings-test.js

```diff
@@ -590,17 +593,169 @@ describe('Convert Message to Placeholder', () => {
   });
 });
 
+describe('collisions', () => {
+  /**
+   * @template {unknown} T
+   * @param {T} input
+   * @return {T}
+   */
+  function deepClone(input) {
+    return JSON.parse(JSON.stringify(input));
+  }
+
+  /** @return {Record<'first'|'second'|'third', CtcWithPlaceholders>} */
```

Comment:
I wonder if we'll be able to use `satisfies` w/ jsdoc... https://devblogs.microsoft.com/typescript/announcing-typescript-4-9-beta/#the-satisfies-operator

Reply:
That would be really nice. I can't make it trigger and the PR doesn't include any jsdoc tests, so I assume it's a different enough parsing flow that it doesn't get picked up in any jsdoc equivalent.